### ISSUE TITLE
mask_path的mask_x4_brain.png路径

### ISSUE THREAD

**Issue by QQ-R7** [lpcccc-cv/MC-VarNet#4]
帅气大哥，请问下这个是自己生成的还是什么，这个该怎么设置啊，帅哥请告诉我QAQ

**Comment by lpcccc-cv** [comment]
hello，重建 mask 的生成代码我已经放代码里了，readme 里边有，超分 mask 生成的代码我随后放 GitHub 上，你也可以自己写一个，很简单的。



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年05月07日 11:47 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***> |
| 主题 | [lpcccc-cv/MC-VarNet] mask_path的mask_x4_brain.png路径 (Issue #4) |

帅气大哥，请问下这个是自己生成的还是什么，这个该怎么设置啊，帅哥请告诉我QAQ

**Comment by QQ-R7** [comment]
谢谢帅哥这么晚还愿意回复我，非常感谢，我想知道是不是自己要把转化后的2D图片再用generate_mask_random.py这个文件生成一系列的mask_x4_.png啊QAQ

**Comment by lpcccc-cv** [comment]
不用的，mask 你可以预先生成好，读数据的时候直接读取 mask 然后对图片做欠采样就行，你也可以把这个代码集成到读数据的文件里，每次读都重新生成 mask



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年05月07日 12:42 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [lpcccc-cv/MC-VarNet] mask_path的mask_x4_brain.png路径 (Issue #4) |

谢谢帅哥这么晚还愿意回复我，非常感谢，我想知道是不是自己要把转化后的2D图片再用generate_mask_random.py这个文件生成一系列的mask_x4_.png啊QAQ

**Comment by QQ-R7** [comment]
谢谢大帅哥，也就是说，我先预生成PD和T2的mask图片，然后对所有的mask欠采样，我还是有些疑问，是分别对IXI数据集的T2和PD进行欠采样，还是将他们一起进行欠采样，因为IXI_dataset.py中mask_path = '/home/lpc/dataset/IXI/MCSR/mask_sr_x4.png'，这里的mask文件路径应该指向哪呀帅哥作者QAQ

**Comment by lpcccc-cv** [comment]
mask_path = '/home/lpc/dataset/IXI/MCSR/mask_sr_x4.png,这个mask是根据你的需求生成的，生成的代码我已经放到github上了，但听你的描述，我感觉你对这个任务的理解可能不是很正确，我论文里有介绍这个任务的退化过程，也就是如何从全采样图像得到欠采样图像，你可以再研究下，有问题可以再问我

**Comment by QQ-R7** [comment]
嗯嗯，谢谢你帅哥

**Comment by qhlyr** [comment]
你好帅哥，我想请问下关于网络的俩个输入，一个是target，还有一个reference，还有一个mask，我看到您网络提示到# x: input low-quality target image [b,1,h,w]
      # y: input high-quality ref image [b,1,h,w]
        # mask: undersampling mask [b,1,h,w]，我想请问您的是，输入的y，高分辨率reference怎么变成和x（lr）一样的尺寸，通过裁剪吗？还是通过其他手段呢，因为我尝试目前的x（lr）是64×64，而y是256×256  T.T

**Comment by QQ-R7** [comment]
超级大帅哥，我理解的论文意思是不是分别要对输入的高分辨率T2和PD进行掩码欠采样啊，但是部分代码我没有理解，VANet.py里网络的前向过程注释说，x是input low-quality target image ，y: input high-quality ref image，这似乎和我的理解有偏差。另外一个问题，我看到配置文件只需要输入高分辨率T2与PD，是不是您的代码会在网络过程中处理HR成LR和掩码，不需要保存在文件夹（虽然我不会，但是你好牛）。最后，我还是不知道IXI.dataset.py中mask_path = '/home/lpc/dataset/IXI/MCSR/mask_sr_x4.png'这里如何设置，您可以举个例子吗？（备注：帅哥你这个框架真牛皮）

**Comment by lpcccc-cv** [comment]
首先，你在跑我的代码之前，比如IXI数据集，要准备好PD和T2的HR以及相应的mask，如果是SRx4的话，就需要。一个x4的mask，也就是mask_path = '/home/lpc/dataset/IXI/MCSR/mask_sr_x4.png'。这个mask的生成方式我已经添加到代码里了。

**Comment by lpcccc-cv** [comment]
``
def gen_mask_for_SR(size, scale):
    h,w = size
    mask = torch.zeros(size)
    lr_h = h//scale
    lr_w = w//scale
    top_left_h = h//2-lr_h//2
    top_left_w = w//2-lr_w//2
    mask[top_left_h:(top_left_h+lr_h), top_left_w:(top_left_w+lr_w)] = torch.ones(lr_h,lr_w)
    return mask
``
也就是这个代码，你可以预先跑下这个代码，把mask存成png或者npy，然后把路径给到mask_path就行了。也可以直接把这个代码嵌入ixi_data.py里，读数据的时候生成这个mask，那也就不需要给这个mask_path 了。

**Comment by QQ-R7** [comment]
谢谢大佬的回复，数据集我准备好了，也按照您论文的方式分为训练验证测试的2D格式，就是掩码我不太确定，您的意思是我将T2和PD生成一系列的mask保存为png格式，然后我填写mask_path = '/home/lpc/dataset/IXI/MCSR/mask_sr_x4.png'，这个路径问题是只有一张mask，那我其他的mask怎么做呢？还是怎么做啊大佬

**Comment by lpcccc-cv** [comment]
就是只有一张 mask 呀



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年05月08日 10:24 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [lpcccc-cv/MC-VarNet] mask_path的mask_x4_brain.png路径 (Issue #4) |

谢谢大佬的回复，数据集我准备好了，也按照您论文的方式分为训练验证测试的2D格式，就是掩码我不太确定，您的意思是我将T2和PD生成一系列的mask保存为png格式，然后我填写mask_path = '/home/lpc/dataset/IXI/MCSR/mask_sr_x4.png'，这个路径问题是只有一张mask，那我其他的mask怎么做呢？还是怎么做啊大佬

**Comment by QQ-R7** [comment]
啊？那我这个mask该怎么进行选择啊？我随便选个HR的T2掩码吗？

**Comment by lpcccc-cv** [comment]
大哥 这个 mask 对所有图片是一样的呀，你只需要一个 mask，就可以对所有图片降采样了，建议你好好了解下这个 task 的背景知识。



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年05月08日 11:10 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [lpcccc-cv/MC-VarNet] mask_path的mask_x4_brain.png路径 (Issue #4) |

啊？那我这个mask该怎么进行选择啊？我随便选个HR的T2掩码吗？

**Comment by QQ-R7** [comment]
怪不得怪不得，我当时就好奇这为啥生成的掩码都一样，然后我试着复现结果结果比你原文的还要好，我就不敢继续下去了，谢谢帅哥愿意回复我，真的感谢，另外我想请问下怎么x2，x4，我看输入的尺寸都256,256，我是否需要将配置文件的crop_size改为比如64（但貌似会出现与mask尺寸不匹配的问题）

**Comment by lpcccc-cv** [comment]
不需要 crop，要不然 跟 mask 尺寸就不一样了



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年05月08日 11:30 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [lpcccc-cv/MC-VarNet] mask_path的mask_x4_brain.png路径 (Issue #4) |

怪不得怪不得，我当时就好奇这为啥生成的掩码都一样，然后我试着复现结果结果比你原文的还要好，我就不敢继续下去了，谢谢帅哥愿意回复我，真的感谢，另外我想请问下怎么x2，x4，我看输入的尺寸都256,256，我是否需要将配置文件的crop_size改为比如64（但貌似会出现与mask尺寸不匹配的问题）

**Comment by QQ-R7** [comment]
哦哦，只需要改scale吗？还要个疑问想问大佬，多对比度的参考图像是用HR还是LR呢？

**Comment by QQ-R7** [comment]
是不是您是通过掩码不同的尺寸来实现不同的x2，x4吗

**Comment by lpcccc-cv** [comment]
是的



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年05月08日 11:47 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [lpcccc-cv/MC-VarNet] mask_path的mask_x4_brain.png路径 (Issue #4) |

是不是您是通过掩码不同的尺寸来实现不同的x2，x4吗

**Comment by QQ-R7** [comment]
我懂了，谢谢你，你是我的神！(・ε・｀)
